### PR TITLE
Adding a page of details for every book when book's title is clicked

### PR DESCRIPTION
**Pull request by AhmadAlkaati**
Fetching all data from online API and making an html page that has details for every book

### CODE REVIEW

**Review by mladenuzelac — APPROVED** [review]
This is nicely done. Solve conflicts and merge it with the main branch.
Pull the code from the repository and create a new branch with a name that related to the task.
New branch should be branched from the main branch.
I see something that will be fixed in the next PR.

**Comment by AhmadAlkaati** [comment]
Hi @mladenuzelac  Thanks for reviewing my code :) 
Should I delete the current pull request and make a new one in a new branch ?

**Comment by mladenuzelac** [comment]
First fix the conflicts and merge into the main branch. 
After the merge create a branch from the main branch.
Name of the branch should reflect what you are doing.
It is harder to fix conflicts when you do a lot of work at once. 
Because you might end in a conflict hell.

File: books/book.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+let currentUrl = new URL(location.href); // creating new url that contains current url
+let searchParams = new URLSearchParams(currentUrl.search); //  searching in current  url params
+let currentUrlId = searchParams.get("id"); //  getting current url id
+window.addEventListener("DomContentLoaded", getBook());
+
+function getBook() {
+  fetch("https://bookshop-api.mirkwood.dev/books/")
```
Comment (josipgrgic) [inline books/book.js:7]: getBook should call the https://bookshop-api.mirkwood.dev/books/id endpoint to get a single book and not all the books. It is not necessary to get all the books just to display info about a single book. Imagine you have 1 million books and used this approach - the user would wait for a long time to get a massive amount of data even though they don't need to.

File: books/book.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    .then((data) => {
+      let main = document.querySelector("main"); // html main element
+      let currentBook = "";
+      data.forEach((book) => {
+        let bookID = book.id; // getting id of every book
+
+        // comparing every book's id with current  url id
+        if (bookID == currentUrlId) {
+          // Getting values of online API and adding them to the UI
+          let title = ` <h2 class = 'title' > ${book.title} </h2>`;
```
Comment (josipgrgic) [inline books/book.js:21]: Create a separate function in which you will handle the display of the book

**Review by josipgrgic — CHANGES_REQUESTED** [review]
Please fix the thing I pointed out and then I will approve the PR :)

**Pull request closed without merging**